assert/assert_greater.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { format } from "./_format.ts";
+import { AssertionError } from "./assertion_error.ts";
+
+/**
+ * Make an assertion that `actual` is greater than `expected`.
+ * If not then throw.
+ */
+export function assertGreater<T>(actual: T, expected: T, msg?: string) {
+  if (actual > expected) return;
+
+  const actualString = format(actual);
+  const expectedString = format(expected);
+  throw new AssertionError(msg ?? `Expect ${actualString} > ${expectedString}`);
```

Comment:
I think we should show both default and given message if `msg` is passed. Maybe let's align to what `assertEquals` does for the given `msg` input. ref. https://github.com/denoland/deno_std/blob/b5c05eea16b542ea6951ef59bfb70a9de0aaca10/assert/assert_equals.ts#L29-L30